rust/lance-encoding/src/encodings/logical/blob.rs

```diff
@@ -327,34 +326,41 @@ impl FieldEncoder for BlobV2StructuralEncoder {
         let mut blob_id_builder = PrimitiveBuilder::<UInt32Type>::with_capacity(binary_array.len());
         let mut uri_builder = StringBuilder::with_capacity(binary_array.len(), 0);
 
-        for i in 0..binary_array.len() {
-            let is_null_row = match array.data_type() {
-                DataType::Struct(_) => array.is_null(i),
-                _ => binary_array.is_null(i),
-            };
-            if is_null_row {
-                kind_builder.append_null();
-                position_builder.append_null();
-                size_builder.append_null();
-                blob_id_builder.append_null();
-                uri_builder.append_null();
+        for i in 0..struct_arr.len() {
+            if struct_arr.is_null(i) {
+                // Packed struct does not support nullable fields; use empty/default values and rely on rep/def.
```

Comment:
Where are the repdef placed?  If I recall correctly we used to place them in the position or size?  Here it looks like you are setting both to zero?